src/state/slices/txHistorySlice/selectors.ts

```diff
@@ -158,28 +185,34 @@ export const selectTxStatusById = createCachedSelector(
 )((_state: ReduxState, txId: TxId) => txId ?? 'undefined')
 
 const selectRebasesById = (state: ReduxState) => state.txHistory.rebases.byId
-export const selectRebasesByAssetId = (state: ReduxState) => state.txHistory.rebases.byAssetId
-export const selectRebaseIdsByAccountId = (state: ReduxState) => state.txHistory.rebases.byAccountId
+export const selectWalletRebaseIdsByAccountIdAssetId = (state: ReduxState) =>
+  state.txHistory.rebases.byAccountIdAssetId
 
 export const selectRebaseIdsByFilter = createDeepEqualOutputSelector(
-  selectRebasesByAssetId,
-  selectRebaseIdsByAccountId,
-  selectAssetIdsParamFromFilter,
-  selectAccountIdsParamFromFilter,
-  (rebasesByAssetId, rebaseIdsByAccountId, assetIds, accountIds) => {
-    // all rebase ids by accountId, may include dupes
-    const rebaseIds = assetIds.map(assetId => rebasesByAssetId[assetId] ?? []).flat()
-    // if we're not filtering on account, return deduped rebase ids for given assets
-    if (!accountIds.length) return Array.from(new Set([...rebaseIds]))
-    const accountRebaseIds = accountIds.map(accountId => rebaseIdsByAccountId[accountId]).flat()
-    return intersection(accountRebaseIds, rebaseIds)
+  selectRebaseIds,
```

Comment:
the plain ids are already sorted - allow us to join this filtered/flattened subset based on the original sort order of all ids